linera-chain/src/manager.rs

```diff
@@ -682,10 +700,13 @@ where
         self.ownership.get().super_owners.contains(owner)
     }
 
-    /// Sets the signed proposal, if it is newer than the known one, and not from a single-leader
-    /// round. Returns whether it was updated.
+    /// Sets the signed proposal, if it is newer than the known one, at most from the first
+    /// single-leader round. Returns whether it was updated.
+    ///
+    /// We don't update the signed proposal for any rounds later than `SingleLeader(0)`,
+    /// because single-leader rounds cannot be skipped without a timeout certificate.
```

Comment:
Why can we skip/update/timeout SingleLeader(0) without the timeout (certificate)?

Reply:
Ah, I see. So the reason we don't return early `false` here for `SingleLeader(0)` isn't b/c it can't be skipped but because it's the first one for which we can update the `self.singed_proposal`?

In such case, I think the `>` condition is not the best test for it, it's not explained by the comment on this function.

Reply:
so maybe simply:
```rust
if matches!(proposal.round, Round::SingleLeader(_)) { return false; }
```
?

Reply:
No, because for `SingleLeader(0)` we want to return `true`, and for later rounds like `Validator` we also want to return `false`.